pageserver/src/virtual_file.rs

```diff
@@ -1155,6 +1158,47 @@ fn get_open_files() -> &'static OpenFiles {
     }
 }
 
+const IO_BUFFER_ALIGNMENT_NOT_SET: usize = 0;
+static IO_BUFFER_ALIGNMENT: AtomicUsize = AtomicUsize::new(IO_BUFFER_ALIGNMENT_NOT_SET);
+
+/// Returns true if `x` is a power of two.
+fn is_power_of_two(x: usize) -> bool {
+    (x != 0) && ((x & (x - 1)) == 0)
+}
+
+#[allow(unused)]
+pub(crate) fn set_io_buffer_alignment(align: usize) -> Result<(), usize> {
+    if is_power_of_two(align) {
+        IO_BUFFER_ALIGNMENT.store(align, std::sync::atomic::Ordering::Relaxed);
+        Ok(())
+    } else {
+        Err(align)
+    }
+}
+
+pub(crate) fn get_io_buffer_alignment() -> usize {
+    let align = IO_BUFFER_ALIGNMENT.load(std::sync::atomic::Ordering::Relaxed);
+
+    if cfg!(test) {
+        let env_var_name = "NEON_PAGESERVER_UNIT_TEST_IO_BUFFER_ALIGNMENT";
```

Comment:
It would be better to somehow run both flavours of unit tests by default. I don't like the fact that these unit tests won't run in CI (too easy to break unless you know of the magic env var).

Reply:
I added the following in the CI matrix run branch for the rust unit tests, which we could add to the mergable PR.

https://github.com/neondatabase/neon/blob/c2dcc1585d9de7f514af6339a64c49b9c5311a74/.github/workflows/_build-and-test-locally.yml#L212-L216

Running the regression tests against both alignments in CI is too costly though.